### DISCUSSION TITLE
PWA vs Apple

### DISCUSSION THREAD

**Discussion by willycamargo** [frontendbr/forum#592] [Perguntas e Respostas]
O Google vem incentivando o desenvolvimento de PWA a algum tempo e esse assunto está em evidência na comunidade a algum tempo. Eu mesmo já estudei bastante e, inclusive, já desenvolvi uma.

Porém, como muitos sabem, a Apple está fazendo o máximo  "frear" essa tecnologia. Obviamente essa tecnologia vai atingir diretamente ao bolso dela, assim como pode atingir também o bolso de Google e Microsoft, porém parece que a Apple realmente esta preocupada com isso.

Acabei de assistir (e ainda estou processando) [um vídeo](https://www.youtube.com/watch?v=FC0pT9xg1oI, 'WWDC 2017 — APPOCALYPSE — Apple') de uma campanha feita pela Apple para incentivar o desenvolvimento de apps nativas. O "problema" desse vídeo é que ele é uma sátira ao que o desenvolvimento de PWA pode levar. IMHO, um anúncio público de que a Apple quer frear essa tecnologia o quanto puder.

Minha reação ao assistir o vídeo pela primeira vez foi de ficar puto, depois apenas dei risada ao mesmo tempo em que chorava. Enfim, o que vocês acham sobre isso? O que tem a dizer?

Threads relacionadas a PWA:

#34 
#470 
#303

**Comment by gabrieldarezzo** [comment]
> O "problema" desse vídeo é que ele é uma sátira

Caramba.. Eu vi o comercial e não tinha pensando nesse aspecto...

A verdade é uma só...

Tem muito App que não precisava ser App... 
Uma  Web App já resolvia lindamente com os recursos novos...

Lembrando que a Google pegou varias pessoas de surpresa com o anuncio dos [Instant Apps](https://developer.android.com/topic/instant-apps/index.html?hl=pt-br) no KeyNote.

**Comment by yanmagale** [comment]
Excelente o tópico @willycamargo .

Nossa, da pra falar horas e horas desse assunto ... Mas vou dar meus 2 centavos na discussão:

- Como você bem disse, o Google tem feito diversas ações e incentivos para que as pessoas desenvolvam PWAs. Existem uma série de estudos, com métricas sobre os usuários, sobre o tempo que ele fica em média nos aplicativos, quantidade de aplicativos usados diariamente, churn, exclusão de apps no seu device, entre outras métricas

- As Stores hoje estão abarrotadas de apps que fazem as mesmas coisas. É díficil se destacar nelas. Muitos destes apps, não necessariamente precisariam de ser aplicações nativas. Até pouco tempo atrás, a web não tinha todo este poder. Hoje, ela tem tanto, mas tanto poder, que podemos fazer tudo o que a nossa imaginação quiser :smile: (Não vou nem comentar sobre a nossa excelente qualidade de Internet)

- Existe uma seríe de feature e conceitos no guarda-chuva das PWAs, mas todas elas visam melhorar a experiência do usuário. Só por este motivo, acredito que todos nós já deveríamos estar atentos a isso.

Sobre o vídeo, sim, a Apple é uma empresa, e comercialmente, faria a receita das stores baixarem muito. Não sei como eles fariam para monetizar estes tipos de apps, ou se iriam criar alguma outra forma de monetização, mas a medida que a comunidade for adotando mais e mais, acredito que é uma solução que eles irão implementar sim.

**Comment by gabrieldarezzo** [comment]
@sergiolopes fez uma palestra muito bacana:

https://www.youtube.com/watch?v=sH7dlRnuh-k

**Comment by leandro1504** [comment]
Prezados,
    Aproveitando o assunto, gostaria de uma indicação de vocês.
    Precisava achar uma solução para push notifications no iphone e ipad, para isso pensei em compilar um app para iphone utilizando o ionic somente para rodar o PWA ? (eu pensei em fazer isso)

Alguma sugestão ?

**Comment by gabrieldarezzo** [comment]
Já fiz em Córdova e foi relativamente fácil.

Dá uma olhada nesse tutorial.
Ele foi difícil de encontrar pelo SEO.

Mas foi um dos melhores:

http://macdonst.github.io/push-workshop/

Tenho um repo chamado: 'cordova_app' pode ajudar caso você seja user de Windows.

A vantagem de fazer utilizando um híbrido é que você iria fazer uma vez e atender todas as plataformas (IOS/Android/web).

**Comment by leandro1504** [comment]
Gabriel,

	Obrigado pela resposta, mas abusando um pouco da sua gentileza, me tira uma outra dúvida, pois sou iniciante neste ramo de desenvolvimento mobile.
	Uma dúvida:
	Existe alguma forma de compilar o App e testar em dispositivos Iphone e Ipad antes de adquirir a licença (assim como é feito no android, onde é gerado o APK, feito o upload para o device e instalado) ?
	
	Desde já agradeço.

**Comment by gabrieldarezzo** [comment]
Você tem um MAC?

Se não me engano é só [baixar  o xCode](https://developer.apple.com/download/) e depois instalar o cordova:
https://cordova.apache.org/docs/en/latest/guide/platforms/ios/index.html

Para testar sem publicação acho que é opcional ter a licença se não me engano.


Vamos ver se alguem do mundo IoS se manifesta

**Comment by malaquiasdev** [comment]
Eu acho que já é meio um caminho sem volta, muita gente já tem enxergado os ganhos de se usar PWA, Twitter e Telegram são fortes exemplos. 

Fico impressionado como a empresa que move tendências está tentando barrar que uma chegue a sua plataforma

**Comment by willycamargo** [comment]
Como alguns disseram, deve estar relacionado diretamente a $.

A Apple move tendências e surpreende seus consumidores quanto a inovações e qualidade. O fato é que esses clientes nem sabem o que é PWA e a Apple ainda não perde nada deixando adotar essas tecnologias em sua plataforma. Pelo contrário, ela pode perder muita grana se adotar.

Não existe pressão dos consumidores, não existe vantagem em implementar, então não irão fazer e irão atrasar o máximo que puder, é isso que consigo entender por enquanto.

**Comment by tcelestino** [comment]
Parece que agora pode acontecer: http://appleinsider.com/articles/17/08/05/apple-may-integrate-service-workers-into-webkit-support-next-gen-web-apps-in-safari

**Comment by willycamargo** [comment]
Eu havia visto esse assunto vagando pelo twitter @tcelestino , mas infelizmente acabei esquecendo de pesquisar mais.

Sensacional, parece que agora teremos mais liberdade e menos medo quanto ao uso de service workers.

**Comment by yanmagale** [comment]
@tcelestino essa notícia é muito importante, que nem o @willycamargo disse, nos passa esta tranquilidade e fico feliz em ver que a Apple ouviu os pedidos da comunidade em prol da evolução da Web

**Comment by ngoctrai1997** [comment]
Existem muitos erros quando atualizei o IOS 11.4.1
Eu não sei o que a Apple quer quando eles limitaram a melhoria da PWA assim
Mas há alguma teia que pode nos ajudar a corrigir esses erros, esse é um: [https://tapita.io/](url)

**Comment by tcelestino** [comment]
A Apple ainda está em fase de implementação do PWA, o que vai levar algum tempo para que o comportamento do Safari esteja bem parecida com a do Chrome no Android. Mas é nítido que  percebi de um tempo pra cá, que na versão preview do Safari, a empresa vem adicionando mais recursos relacionadas ao PWA.

Recomendo ler esse [artigo](https://medium.com/@firt/progressive-web-apps-on-ios-are-here-d00430dee3a7) de março que mostra um pouco das "deficiências" em relação ao PWA no Safari.
